Index: gnome-games-2.27.4/aisleriot/window.c
===================================================================
--- gnome-games-2.27.4.orig/aisleriot/window.c	2009-07-14 05:55:38.000000000 +1000
+++ gnome-games-2.27.4/aisleriot/window.c	2009-07-14 09:04:28.000000000 +1000
@@ -29,6 +29,7 @@
 
 #include <gdk/gdk.h>
 #include <gtk/gtk.h>
+#include <launchpad-integration.h>
 
 #ifdef HAVE_HILDON
 #ifdef HAVE_MAEMO_3
@@ -2432,6 +2433,7 @@
         "<menu action='HelpMenu'>"
           "<menuitem action='Contents'/>"
           "<menuitem action='HelpGame'/>"
+          "<placeholder name='LaunchpadItems'/>"
           "<menuitem action='About'/>"
         "</menu>"
 #ifdef ENABLE_DEBUG_UI
@@ -2677,6 +2679,9 @@
   accel_group = gtk_ui_manager_get_accel_group (priv->ui_manager);
   gtk_window_add_accel_group (GTK_WINDOW (window), accel_group);
 
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (priv->ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+ 
   action = gtk_action_group_get_action (priv->action_group, "Toolbar");
   priv->toolbar_visible = games_conf_get_boolean (NULL, aisleriot_conf_get_key (CONF_SHOW_TOOLBAR), NULL) != FALSE;
   gtk_toggle_action_set_active (GTK_TOGGLE_ACTION (action),
Index: gnome-games-2.27.4/blackjack/src/blackjack.cpp
===================================================================
--- gnome-games-2.27.4.orig/blackjack/src/blackjack.cpp	2009-07-14 05:55:38.000000000 +1000
+++ gnome-games-2.27.4/blackjack/src/blackjack.cpp	2009-07-14 09:04:28.000000000 +1000
@@ -26,6 +26,7 @@
 #include <string.h>
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
+#include <launchpad-integration.h>
 
 #include <libgames-support/games-conf.h>
 #include <libgames-support/games-stock.h>
@@ -314,6 +315,7 @@
                 "    </menu>"
                 "    <menu name='HelpMenu' action='help-menu'>"
                 "      <menuitem name='Contents' action='show-help-contents'/>"
+                "      <placeholder name='LaunchpadItems'/>"
                 "      <menuitem name='About' action='show-about'/>"
                 "    </menu>"
                 "  </menubar>"
@@ -366,6 +368,9 @@
 
         gtk_ui_manager_add_ui_from_string (ui, ui_definition, -1, &error);
 
+        launchpad_integration_set_sourcepackagename("gnome-games");
+        launchpad_integration_add_ui (ui, "/MenuBar/HelpMenu/LaunchpadItems");
+
         actions = gtk_action_group_new ("Actions");
         gtk_action_group_set_translation_domain (actions, GETTEXT_PACKAGE);
         gtk_action_group_add_actions (actions, entries, G_N_ELEMENTS (entries), toplevel_window);
Index: gnome-games-2.27.4/configure.in
===================================================================
--- gnome-games-2.27.4.orig/configure.in	2009-07-14 06:24:31.000000000 +1000
+++ gnome-games-2.27.4/configure.in	2009-07-14 09:04:28.000000000 +1000
@@ -587,7 +587,8 @@
 # Check for GNOME modules
 
 if test "$with_platform" = "gnome"; then
-  PKG_CHECK_MODULES([GNOME],[gconf-2.0 >= $GCONF_REQUIRED])
+  PKG_CHECK_MODULES([GNOME],[gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration])
   AC_SUBST([GNOME_CFLAGS])
   AC_SUBST([GNOME_LIBS])
 
Index: gnome-games-2.27.4/glines/glines.c
===================================================================
--- gnome-games-2.27.4.orig/glines/glines.c	2009-07-14 05:55:38.000000000 +1000
+++ gnome-games-2.27.4/glines/glines.c	2009-07-14 09:04:28.000000000 +1000
@@ -33,6 +33,7 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
+#include <launchpad-integration.h>
 #include <gdk/gdkkeysyms.h>
 
 #include <libgames-support/games-conf.h>
@@ -1722,6 +1723,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>" "    </menu>" "  </menubar>" "</ui>";
 
 static void
@@ -1746,6 +1748,9 @@
   accel_group = gtk_ui_manager_get_accel_group (ui_manager);
   gtk_window_add_accel_group (GTK_WINDOW (app), accel_group);
 
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   scoreitem = gtk_ui_manager_get_widget (ui_manager,
                                          "/MainMenu/GameMenu/Scores");
   menubar = gtk_ui_manager_get_widget (ui_manager, "/MainMenu");
Index: gnome-games-2.27.4/gnect/src/main.c
===================================================================
--- gnome-games-2.27.4.orig/gnect/src/main.c	2009-07-14 05:55:38.000000000 +1000
+++ gnome-games-2.27.4/gnect/src/main.c	2009-07-14 09:04:28.000000000 +1000
@@ -27,6 +27,7 @@
 
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
+#include <launchpad-integration.h>
 
 #include <libgames-support/games-conf.h>
 #include <libgames-support/games-gridframe.h>
@@ -1344,6 +1345,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>" "    </menu>" "  </menubar>" "</ui>";
 
 
@@ -1363,6 +1365,9 @@
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
 
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   gtk_window_add_accel_group (GTK_WINDOW (app),
 			      gtk_ui_manager_get_accel_group (ui_manager));
 
Index: gnome-games-2.27.4/gnibbles/main.c
===================================================================
--- gnome-games-2.27.4.orig/gnibbles/main.c	2009-07-14 05:55:38.000000000 +1000
+++ gnome-games-2.27.4/gnibbles/main.c	2009-07-14 09:04:28.000000000 +1000
@@ -28,6 +28,7 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
+#include <launchpad-integration.h>
 
 #include <libgames-support/games-conf.h>
 #include <libgames-support/games-gridframe.h>
@@ -818,6 +819,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>" "    </menu>" "  </menubar>" "</ui>";
 
 static void
@@ -834,6 +836,9 @@
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
 
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   new_game_action = gtk_action_group_get_action (action_group, "NewGame");
   scores_action = gtk_action_group_get_action (action_group, "Scores");
   end_game_action = gtk_action_group_get_action (action_group, "EndGame");
Index: gnome-games-2.27.4/gnobots2/menu.c
===================================================================
--- gnome-games-2.27.4.orig/gnobots2/menu.c	2009-07-14 05:55:38.000000000 +1000
+++ gnome-games-2.27.4/gnobots2/menu.c	2009-07-14 09:04:28.000000000 +1000
@@ -25,6 +25,7 @@
 
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
+#include <launchpad-integration.h>
 
 #include <libgames-support/games-help.h>
 #include <libgames-support/games-stock.h>
@@ -123,6 +124,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>"
   "    </menu>"
   "  </menubar>"
@@ -355,6 +357,9 @@
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
 
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   scores_action = gtk_action_group_get_action (action_group, "Scores");
   teleport_action = gtk_action_group_get_action (action_group, "Teleport");
   random_action = gtk_action_group_get_action (action_group, "Random");
Index: gnome-games-2.27.4/gnometris/tetris.cpp
===================================================================
--- gnome-games-2.27.4.orig/gnometris/tetris.cpp	2009-07-14 05:55:38.000000000 +1000
+++ gnome-games-2.27.4/gnometris/tetris.cpp	2009-07-14 09:04:28.000000000 +1000
@@ -30,6 +30,7 @@
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 #include <gio/gio.h>
+#include <launchpad-integration.h>
 
 #include <libgames-support/games-controls.h>
 #include <libgames-support/games-frame.h>
@@ -138,6 +139,7 @@
 	"    </menu>"
 	"    <menu action='HelpMenu'>"
 	"      <menuitem action='Contents'/>"
+        "      <placeholder name='LaunchpadItems'/>"
 	"      <menuitem action='About'/>"
 	"    </menu>"
 	"  </menubar>"
@@ -189,6 +191,10 @@
 	ui_manager = gtk_ui_manager_new ();
 	gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
 	gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
+
+        launchpad_integration_set_sourcepackagename("gnome-games");
+        launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
 	accel_group = gtk_ui_manager_get_accel_group (ui_manager);
 	gtk_window_add_accel_group (GTK_WINDOW (w), accel_group);
 
Index: gnome-games-2.27.4/gnomine/gnomine.c
===================================================================
--- gnome-games-2.27.4.orig/gnomine/gnomine.c	2009-07-14 05:55:38.000000000 +1000
+++ gnome-games-2.27.4/gnomine/gnomine.c	2009-07-14 09:04:28.000000000 +1000
@@ -31,6 +31,7 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
+#include <launchpad-integration.h>
 
 #include <libgames-support/games-clock.h>
 #include <libgames-support/games-conf.h>
@@ -866,6 +867,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>" "    </menu>" "  </menubar>" "</ui>";
 
 static GtkUIManager *
@@ -882,6 +884,10 @@
   ui_manager = gtk_ui_manager_new ();
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
+
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   hint_action = gtk_action_group_get_action (action_group, "Hint");
 
   fullscreen_action =
Index: gnome-games-2.27.4/gnotski/gnotski.c
===================================================================
--- gnome-games-2.27.4.orig/gnotski/gnotski.c	2009-07-14 05:55:39.000000000 +1000
+++ gnome-games-2.27.4/gnotski/gnotski.c	2009-07-14 09:04:28.000000000 +1000
@@ -24,6 +24,7 @@
 #include <string.h>
 #include <stdlib.h>
 
+#include <launchpad-integration.h>
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
@@ -458,6 +459,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>" "    </menu>" "  </menubar>" "</ui>";
 
 /* Session Options */
@@ -1083,6 +1085,10 @@
   ui_manager = gtk_ui_manager_new ();
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
+
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   add_puzzle_menu (ui_manager);
 
   accel_group = gtk_ui_manager_get_accel_group (ui_manager);
Index: gnome-games-2.27.4/gtali/gyahtzee.c
===================================================================
--- gnome-games-2.27.4.orig/gtali/gyahtzee.c	2009-07-14 05:55:39.000000000 +1000
+++ gnome-games-2.27.4/gtali/gyahtzee.c	2009-07-14 09:04:28.000000000 +1000
@@ -43,6 +43,7 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
+#include <launchpad-integration.h>
 
 #include <libgames-support/games-help.h>
 #include <libgames-support/games-stock.h>
@@ -731,6 +732,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>"
   "    </menu>" "  </menubar>" "  <accelerator action='Roll' />" "</ui>";
 
@@ -748,6 +750,10 @@
 
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
+
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   scores_action = gtk_action_group_get_action (action_group, "Scores");
   undo_action   = gtk_action_group_get_action (action_group, "Undo");
   update_undo_sensitivity();
Index: gnome-games-2.27.4/iagno/gnothello.c
===================================================================
--- gnome-games-2.27.4.orig/iagno/gnothello.c	2009-07-14 05:55:39.000000000 +1000
+++ gnome-games-2.27.4/iagno/gnothello.c	2009-07-14 09:04:28.000000000 +1000
@@ -47,6 +47,8 @@
 #include <ggz-embed.h>
 #endif /* WITH GGZ_CLIENT */
 
+#include <launchpad-integration.h>
+
 #include "gnothello.h"
 #include "othello.h"
 #include "properties.h"
@@ -850,6 +852,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>"
   "    </menu>"
   "  </menubar>"
@@ -940,6 +943,9 @@
   statusbar = gtk_statusbar_new ();
   gtk_box_pack_start (GTK_BOX (vbox), statusbar, FALSE, FALSE, 0);
 
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   table = gtk_table_new (1, 8, FALSE);
 
   black_score = gtk_label_new (_("Dark:"));
Index: gnome-games-2.27.4/mahjongg/mahjongg.c
===================================================================
--- gnome-games-2.27.4.orig/mahjongg/mahjongg.c	2009-07-14 05:55:39.000000000 +1000
+++ gnome-games-2.27.4/mahjongg/mahjongg.c	2009-07-14 09:04:28.000000000 +1000
@@ -24,6 +24,7 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
+#include <launchpad-integration.h>
 
 #include <libgames-support/games-clock.h>
 #include <libgames-support/games-conf.h>
@@ -1300,6 +1301,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>"
   "    </menu>"
   "  </menubar>"
@@ -1332,6 +1334,10 @@
 
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
+
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   restart_action = gtk_action_group_get_action (action_group, "RestartGame");
   pause_action = gtk_action_group_get_action (action_group, "PauseGame");
   resume_action = gtk_action_group_get_action (action_group, "ResumeGame");
Index: gnome-games-2.27.4/same-gnome/ui.c
===================================================================
--- gnome-games-2.27.4.orig/same-gnome/ui.c	2009-07-14 05:55:39.000000000 +1000
+++ gnome-games-2.27.4/same-gnome/ui.c	2009-07-14 09:04:28.000000000 +1000
@@ -11,6 +11,8 @@
 #include <gtk/gtk.h>
 #include <glib/gi18n.h>
 
+#include <launchpad-integration.h>
+
 #include <libgames-support/games-conf.h>
 #include <libgames-support/games-files.h>
 #include <libgames-support/games-gridframe.h>
@@ -510,6 +512,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>"
   "    </menu>"
   "  </menubar>"
@@ -569,6 +572,9 @@
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 1);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
 
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   gtk_window_add_accel_group (GTK_WINDOW (application),
 			      gtk_ui_manager_get_accel_group (ui_manager));
 
Index: gnome-games-2.27.4/glchess/src/lib/gtkui/gtkui.py
===================================================================
--- gnome-games-2.27.4.orig/glchess/src/lib/gtkui/gtkui.py	2009-07-14 05:55:38.000000000 +1000
+++ gnome-games-2.27.4/glchess/src/lib/gtkui/gtkui.py	2009-07-14 09:04:28.000000000 +1000
@@ -21,6 +21,7 @@
 import pango
 
 from glchess.defaults import *
+import LaunchpadIntegration
 
 # Stop PyGTK from catching exceptions
 os.environ['PYGTK_FATAL_EXCEPTIONS'] = '1'
@@ -219,6 +220,11 @@
         cell = gtk.CellRendererText()
         combo.pack_start(cell, False)
         combo.add_attribute(cell, 'text', 2)
+
+        # Add launchpad integration
+        widget = self._gui.get_object('help2_menu')
+        LaunchpadIntegration.set_sourcepackagename('gnome-games')
+        LaunchpadIntegration.add_items(widget,1,True,False)	
         
         self._updateViewButtons()
         
Index: gnome-games-2.27.4/gnome-sudoku/src/lib/main.py
===================================================================
--- gnome-games-2.27.4.orig/gnome-sudoku/src/lib/main.py	2009-07-14 05:55:38.000000000 +1000
+++ gnome-games-2.27.4/gnome-sudoku/src/lib/main.py	2009-07-14 09:04:28.000000000 +1000
@@ -25,6 +25,7 @@
         IMAGE_DIR, LICENSE, MIN_NEW_PUZZLES, UI_DIR, VERSION, WEBSITE, WEBSITE_LABEL)
 from gtk_goodies import gconf_wrapper, Undo, dialog_extras
 from simple_debug import simple_debug, options
+import LaunchpadIntegration
 
 ICON_FACTORY = gtk.IconFactory()
 STOCK_PIXBUFS = {}
@@ -121,6 +122,7 @@
         </menu>
       <menu action="Help">
         <menuitem action="ShowHelp"/>
+        <placeholder name='LaunchpadItems'/>
         <menuitem action="About"/>
       </menu>
     </menubar>
@@ -222,6 +224,9 @@
         self.setup_undo()
         self.setup_autosave()
         self.w.add_accel_group(self.uimanager.get_accel_group())
+        # Add launchpad integration
+        LaunchpadIntegration.set_sourcepackagename("gnome-games")
+        LaunchpadIntegration.add_ui(self.uimanager, "/MenuBar/Help/LaunchpadItems")
         self.setup_main_boxes()
         self.setup_tracker_interface()
         self.setup_toggles()
