Index: gnome-games-2.23.3/aisleriot/window.c
===================================================================
--- gnome-games-2.23.3.orig/aisleriot/window.c	2008-06-02 19:45:32.000000000 +0200
+++ gnome-games-2.23.3/aisleriot/window.c	2008-06-11 10:53:35.000000000 +0200
@@ -49,6 +49,8 @@
 #endif
 #endif /* HAVE_HILDON */
 
+#include <launchpad-integration.h>
+
 #include <libgames-support/games-clock.h>
 #include <libgames-support/games-files.h>
 #include <libgames-support/games-stock.h>
@@ -2275,6 +2277,7 @@
         "<menu action='HelpMenu'>"
           "<menuitem action='Contents'/>"
           "<menuitem action='HelpGame'/>"
+          "<placeholder name='LaunchpadItems'/>"
           "<menuitem action='About'/>"
         "</menu>"
 #ifdef ENABLE_DEBUG_UI
@@ -2504,6 +2507,9 @@
   accel_group = gtk_ui_manager_get_accel_group (priv->ui_manager);
   gtk_window_add_accel_group (GTK_WINDOW (window), accel_group);
 
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (priv->ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+ 
   action = gtk_action_group_get_action (priv->action_group, "Toolbar");
   priv->toolbar_visible = games_conf_get_boolean (NULL, aisleriot_conf_get_key (CONF_SHOW_TOOLBAR), NULL) != FALSE;
   gtk_toggle_action_set_active (GTK_TOGGLE_ACTION (action),
Index: gnome-games-2.23.3/blackjack/src/blackjack.cpp
===================================================================
--- gnome-games-2.23.3.orig/blackjack/src/blackjack.cpp	2008-06-02 19:45:32.000000000 +0200
+++ gnome-games-2.23.3/blackjack/src/blackjack.cpp	2008-06-11 10:53:35.000000000 +0200
@@ -25,6 +25,7 @@
 #include <stdlib.h>
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
+#include <launchpad-integration.h>
 #include <games-conf.h>
 #include <games-stock.h>
 
@@ -315,6 +316,7 @@
                 "    </menu>"
                 "    <menu name='HelpMenu' action='help-menu'>"
                 "      <menuitem name='Contents' action='show-help-contents'/>"
+                "      <placeholder name='LaunchpadItems'/>"
                 "      <menuitem name='About' action='show-about'/>"
                 "    </menu>"
                 "  </menubar>"
@@ -367,6 +369,9 @@
 
         gtk_ui_manager_add_ui_from_string (ui, ui_definition, -1, &error);
 
+        launchpad_integration_set_sourcepackagename("gnome-games");
+        launchpad_integration_add_ui (ui, "/MenuBar/HelpMenu/LaunchpadItems");
+
         actions = gtk_action_group_new ("Actions");
         gtk_action_group_set_translation_domain (actions, GETTEXT_PACKAGE);
         gtk_action_group_add_actions (actions, entries, G_N_ELEMENTS (entries), toplevel_window);
Index: gnome-games-2.23.3/configure.in
===================================================================
--- gnome-games-2.23.3.orig/configure.in	2008-06-02 20:24:03.000000000 +0200
+++ gnome-games-2.23.3/configure.in	2008-06-11 10:53:35.000000000 +0200
@@ -366,7 +366,8 @@
   PKG_CHECK_MODULES([GNOME],[
     libxml-2.0 >= $LIBXML_REQUIRED
     libgnome-2.0 >= $LIBGNOME_REQUIRED
-    libgnomeui-2.0 >= $LIBGNOMEUI_REQUIRED])
+    libgnomeui-2.0 >= $LIBGNOMEUI_REQUIRED
+    launchpad-integration])
   AC_SUBST([GNOME_CFLAGS])
   AC_SUBST([GNOME_LIBS])
 
Index: gnome-games-2.23.3/glines/glines.c
===================================================================
--- gnome-games-2.23.3.orig/glines/glines.c	2008-05-29 17:01:50.000000000 +0200
+++ gnome-games-2.23.3/glines/glines.c	2008-06-11 10:53:35.000000000 +0200
@@ -32,6 +32,7 @@
 #include <glib.h>
 #include <glib/gi18n.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
+#include <launchpad-integration.h>
 #include <gdk/gdkkeysyms.h>
 #include <games-scores.h>
 #include <games-scores-dialog.h>
@@ -1732,6 +1733,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>" "    </menu>" "  </menubar>" "</ui>";
 
 static void
@@ -1756,6 +1758,9 @@
   accel_group = gtk_ui_manager_get_accel_group (ui_manager);
   gtk_window_add_accel_group (GTK_WINDOW (app), accel_group);
 
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   scoreitem = gtk_ui_manager_get_widget (ui_manager,
 					 "/MainMenu/GameMenu/Scores");
   menubar = gtk_ui_manager_get_widget (ui_manager, "/MainMenu");
Index: gnome-games-2.23.3/gnect/src/main.c
===================================================================
--- gnome-games-2.23.3.orig/gnect/src/main.c	2008-05-29 17:02:05.000000000 +0200
+++ gnome-games-2.23.3/gnect/src/main.c	2008-06-11 10:53:35.000000000 +0200
@@ -26,6 +26,7 @@
 
 #include "config.h"
 #include <gnome.h>
+#include <launchpad-integration.h>
 
 #include <games-gridframe.h>
 #include <games-stock.h>
@@ -1341,6 +1342,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>" "    </menu>" "  </menubar>" "</ui>";
 
 
@@ -1360,6 +1362,9 @@
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
 
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   gtk_window_add_accel_group (GTK_WINDOW (app),
 			      gtk_ui_manager_get_accel_group (ui_manager));
 
Index: gnome-games-2.23.3/gnibbles/main.c
===================================================================
--- gnome-games-2.23.3.orig/gnibbles/main.c	2008-05-29 17:01:53.000000000 +0200
+++ gnome-games-2.23.3/gnibbles/main.c	2008-06-11 10:53:35.000000000 +0200
@@ -24,6 +24,7 @@
 #include <string.h>
 #include <gdk/gdkkeysyms.h>
 #include <time.h>
+#include <launchpad-integration.h>
 
 #include <games-gridframe.h>
 #include <games-stock.h>
@@ -829,6 +830,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>" "    </menu>" "  </menubar>" "</ui>";
 
 static void
@@ -845,6 +847,9 @@
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
 
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   new_game_action = gtk_action_group_get_action (action_group, "NewGame");
   scores_action = gtk_action_group_get_action (action_group, "Scores");
   end_game_action = gtk_action_group_get_action (action_group, "EndGame");
Index: gnome-games-2.23.3/gnobots2/menu.c
===================================================================
--- gnome-games-2.23.3.orig/gnobots2/menu.c	2008-05-29 17:02:05.000000000 +0200
+++ gnome-games-2.23.3/gnobots2/menu.c	2008-06-11 10:53:35.000000000 +0200
@@ -25,6 +25,7 @@
 #include <games-stock.h>
 #include <games-scores.h>
 #include <games-scores-dialog.h>
+#include <launchpad-integration.h>
 
 #include "gbdefs.h"
 #include "menu.h"
@@ -118,6 +119,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>"
   "    </menu>"
   "  </menubar>"
@@ -350,6 +352,9 @@
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
 
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   scores_action = gtk_action_group_get_action (action_group, "Scores");
   teleport_action = gtk_action_group_get_action (action_group, "Teleport");
   random_action = gtk_action_group_get_action (action_group, "Random");
Index: gnome-games-2.23.3/gnometris/tetris.cpp
===================================================================
--- gnome-games-2.23.3.orig/gnometris/tetris.cpp	2008-05-29 17:01:34.000000000 +0200
+++ gnome-games-2.23.3/gnometris/tetris.cpp	2008-06-11 10:53:35.000000000 +0200
@@ -36,6 +36,7 @@
 
 #include <libgnomevfs/gnome-vfs.h>
 
+#include <launchpad-integration.h>
 #include <gdk/gdkkeysyms.h>
 #include <config.h>
 #include <dirent.h>
@@ -134,6 +135,7 @@
 	"    </menu>"
 	"    <menu action='HelpMenu'>"
 	"      <menuitem action='Contents'/>"
+        "      <placeholder name='LaunchpadItems'/>"
 	"      <menuitem action='About'/>"
 	"    </menu>"
 	"  </menubar>"
@@ -186,6 +188,10 @@
 	ui_manager = gtk_ui_manager_new ();
 	gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
 	gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
+
+        launchpad_integration_set_sourcepackagename("gnome-games");
+        launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
 	accel_group = gtk_ui_manager_get_accel_group (ui_manager);
 	gtk_window_add_accel_group (GTK_WINDOW (w), accel_group);
 
Index: gnome-games-2.23.3/gnomine/gnomine.c
===================================================================
--- gnome-games-2.23.3.orig/gnomine/gnomine.c	2008-05-29 17:01:30.000000000 +0200
+++ gnome-games-2.23.3/gnomine/gnomine.c	2008-06-11 10:53:35.000000000 +0200
@@ -27,6 +27,7 @@
 #include <gnome.h>
 #include <glib/gi18n.h>
 #include <string.h>
+#include <launchpad-integration.h>
 /*#include <gst/gst.h>*/
 
 #include "minefield.h"
@@ -842,6 +843,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>" "    </menu>" "  </menubar>" "</ui>";
 
 static GtkUIManager *
@@ -858,6 +860,10 @@
   ui_manager = gtk_ui_manager_new ();
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
+
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   hint_action = gtk_action_group_get_action (action_group, "Hint");
 
   fullscreen_action =
Index: gnome-games-2.23.3/gnotski/gnotski.c
===================================================================
--- gnome-games-2.23.3.orig/gnotski/gnotski.c	2008-06-02 19:45:33.000000000 +0200
+++ gnome-games-2.23.3/gnotski/gnotski.c	2008-06-11 10:53:35.000000000 +0200
@@ -24,6 +24,7 @@
 #include <string.h>
 #include <stdlib.h>
 
+#include <launchpad-integration.h>
 #include <glib/gi18n.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #include <gtk/gtk.h>
@@ -462,6 +463,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>" "    </menu>" "  </menubar>" "</ui>";
 
 /* Session Options */
@@ -1112,6 +1114,10 @@
   ui_manager = gtk_ui_manager_new ();
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
+
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   add_puzzle_menu (ui_manager);
 
   accel_group = gtk_ui_manager_get_accel_group (ui_manager);
Index: gnome-games-2.23.3/gtali/gyahtzee.c
===================================================================
--- gnome-games-2.23.3.orig/gtali/gyahtzee.c	2008-05-29 17:01:32.000000000 +0200
+++ gnome-games-2.23.3/gtali/gyahtzee.c	2008-06-11 10:53:35.000000000 +0200
@@ -39,6 +39,7 @@
 
 #include <config.h>
 #include <gnome.h>
+#include <launchpad-integration.h>
 
 #include <games-stock.h>
 #include <games-scores.h>
@@ -731,6 +732,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>"
   "    </menu>" "  </menubar>" "  <accelerator action='Roll' />" "</ui>";
 
@@ -748,6 +750,10 @@
 
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
+
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   scores_action = gtk_action_group_get_action (action_group, "Scores");
   undo_action   = gtk_action_group_get_action (action_group, "Undo");
   update_undo_sensitivity();
Index: gnome-games-2.23.3/iagno/gnothello.c
===================================================================
--- gnome-games-2.23.3.orig/iagno/gnothello.c	2008-05-29 17:02:04.000000000 +0200
+++ gnome-games-2.23.3/iagno/gnothello.c	2008-06-11 10:53:35.000000000 +0200
@@ -38,6 +38,8 @@
 #include <ggz-embed.h>
 #endif
 
+#include <launchpad-integration.h>
+
 #include "gnothello.h"
 #include "othello.h"
 #include "properties.h"
@@ -659,6 +661,9 @@
   gnome_app_set_statusbar (GNOME_APP (window), GTK_WIDGET (appbar));
   gnome_app_install_menu_hints (GNOME_APP (window), mainmenu);
 
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_items (gtk_menu_item_get_submenu (((GnomeUIInfo)mainmenu[2]).widget), -1, TRUE, TRUE);
+
   table = gtk_table_new (1, 8, FALSE);
 
   black_score = gtk_label_new (_("Dark:"));
Index: gnome-games-2.23.3/mahjongg/mahjongg.c
===================================================================
--- gnome-games-2.23.3.orig/mahjongg/mahjongg.c	2008-06-02 19:45:33.000000000 +0200
+++ gnome-games-2.23.3/mahjongg/mahjongg.c	2008-06-11 10:53:35.000000000 +0200
@@ -25,6 +25,7 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
+#include <launchpad-integration.h>
 
 #include <libgames-support/games-clock.h>
 #include <libgames-support/games-conf.h>
@@ -1332,6 +1333,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>"
   "    </menu>"
   "  </menubar>"
@@ -1364,6 +1366,10 @@
 
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 0);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
+
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   restart_action = gtk_action_group_get_action (action_group, "RestartGame");
   pause_action = gtk_action_group_get_action (action_group, "PauseGame");
   resume_action = gtk_action_group_get_action (action_group, "ResumeGame");
Index: gnome-games-2.23.3/same-gnome/ui.c
===================================================================
--- gnome-games-2.23.3.orig/same-gnome/ui.c	2008-05-29 17:01:57.000000000 +0200
+++ gnome-games-2.23.3/same-gnome/ui.c	2008-06-11 10:53:35.000000000 +0200
@@ -15,6 +15,8 @@
 #include <libgnome/gnome-help.h>
 #endif /* HAVE_GNOME */
 
+#include <launchpad-integration.h>
+
 #include <libgames-support/games-conf.h>
 #include <libgames-support/games-files.h>
 #include <libgames-support/games-gridframe.h>
@@ -518,6 +520,7 @@
   "    </menu>"
   "    <menu action='HelpMenu'>"
   "      <menuitem action='Contents'/>"
+  "      <placeholder name='LaunchpadItems'/>"
   "      <menuitem action='About'/>"
   "    </menu>"
   "  </menubar>"
@@ -577,6 +580,9 @@
   gtk_ui_manager_insert_action_group (ui_manager, action_group, 1);
   gtk_ui_manager_add_ui_from_string (ui_manager, ui_description, -1, NULL);
 
+  launchpad_integration_set_sourcepackagename("gnome-games");
+  launchpad_integration_add_ui (ui_manager, "/MainMenu/HelpMenu/LaunchpadItems");
+
   gtk_window_add_accel_group (GTK_WINDOW (application),
 			      gtk_ui_manager_get_accel_group (ui_manager));
 
