
Index: gnome-games-2.25.90/configure
===================================================================
--- gnome-games-2.25.90.orig/configure	2009-02-04 11:46:50.000000000 +0100
+++ gnome-games-2.25.90/configure	2009-02-04 11:48:22.000000000 +0100
@@ -21195,12 +21195,15 @@
         pkg_cv_GNOME_CFLAGS="$GNOME_CFLAGS"
     else
         if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gconf-2.0 >= \$GCONF_REQUIRED\"") >&5
-  ($PKG_CONFIG --exists --print-errors "gconf-2.0 >= $GCONF_REQUIRED") 2>&5
+    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gconf-2.0 >= \$GCONF_REQUIRED
+                             launchpad-integration\"") >&5
+  ($PKG_CONFIG --exists --print-errors "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_GNOME_CFLAGS=`$PKG_CONFIG --cflags "gconf-2.0 >= $GCONF_REQUIRED" 2>/dev/null`
+  pkg_cv_GNOME_CFLAGS=`$PKG_CONFIG --cflags "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -21213,12 +21216,15 @@
         pkg_cv_GNOME_LIBS="$GNOME_LIBS"
     else
         if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gconf-2.0 >= \$GCONF_REQUIRED\"") >&5
-  ($PKG_CONFIG --exists --print-errors "gconf-2.0 >= $GCONF_REQUIRED") 2>&5
+    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gconf-2.0 >= \$GCONF_REQUIRED
+                             launchpad-integration\"") >&5
+  ($PKG_CONFIG --exists --print-errors "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_GNOME_LIBS=`$PKG_CONFIG --libs "gconf-2.0 >= $GCONF_REQUIRED" 2>/dev/null`
+  pkg_cv_GNOME_LIBS=`$PKG_CONFIG --libs "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -21237,14 +21243,17 @@
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        GNOME_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "gconf-2.0 >= $GCONF_REQUIRED"`
+	        GNOME_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration"`
         else
-	        GNOME_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "gconf-2.0 >= $GCONF_REQUIRED"`
+	        GNOME_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration"`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$GNOME_PKG_ERRORS" >&5
 
-	{ { echo "$as_me:$LINENO: error: Package requirements (gconf-2.0 >= $GCONF_REQUIRED) were not met:
+	{ { echo "$as_me:$LINENO: error: Package requirements (gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration) were not met:
 
 $GNOME_PKG_ERRORS
 
@@ -21255,7 +21264,8 @@
 and GNOME_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details.
 " >&5
-echo "$as_me: error: Package requirements (gconf-2.0 >= $GCONF_REQUIRED) were not met:
+echo "$as_me: error: Package requirements (gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration) were not met:
 
 $GNOME_PKG_ERRORS
 
@@ -23338,7 +23348,7 @@
 do
   for j in $libggz_header;
   do
-    echo "configure: 23341: $i/$j" >&5
+    echo "configure: 23351: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       libggz_incdir=$i
@@ -23609,7 +23619,7 @@
 do
   for j in $ggzmod_header;
   do
-    echo "configure: 23612: $i/$j" >&5
+    echo "configure: 23622: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       ggzmod_incdir=$i
@@ -23749,7 +23759,7 @@
 do
   for j in ggz-config;
   do
-    echo "configure: 23752: $i/$j" >&5
+    echo "configure: 23762: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       ggz_config_dir=$i
@@ -23967,7 +23977,7 @@
 do
   for j in $libggz_header;
   do
-    echo "configure: 23970: $i/$j" >&5
+    echo "configure: 23980: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       libggz_incdir=$i
@@ -24254,7 +24264,7 @@
 do
   for j in $ggzdmod_header;
   do
-    echo "configure: 24257: $i/$j" >&5
+    echo "configure: 24267: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       ggzdmod_incdir=$i
@@ -25349,7 +25359,7 @@
 do
   for j in $gnutls_header;
   do
-    echo "configure: 25352: $i/$j" >&5
+    echo "configure: 25362: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       gnutls_incdir=$i
@@ -25440,7 +25450,7 @@
 do
   for j in openssl/ssl.h;
   do
-    echo "configure: 25443: $i/$j" >&5
+    echo "configure: 25453: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       ssl_incdir=$i
