Index: gnome-games-2.27.2/gnometris/gnometris.schemas.in
===================================================================
--- gnome-games-2.27.2.orig/gnometris/gnometris.schemas.in	2009-05-25 21:39:29.000000000 +0200
+++ gnome-games-2.27.2/gnometris/gnometris.schemas.in	2009-05-26 10:27:29.000000000 +0200
@@ -18,7 +18,7 @@
       <applyto>/apps/gnometris/options/theme</applyto>
       <owner>gnometris</owner>
       <type>string</type> 
-      <default>tangoshaded</default>
+      <default>plain</default>
       <locale name="C">
         <short>The theme used for rendering the blocks</short>
         <long>The name of the theme used for rendering the blocks and the background.</long>
