diff -Nur gnome-games-2.26.1/configure gnome-games-2.26.1.ubuntu/configure
--- gnome-games-2.26.1/configure	2009-04-14 05:07:07.000000000 +0200
+++ gnome-games-2.26.1.ubuntu/configure	2009-04-14 12:52:37.000000000 +0200
@@ -20964,7 +20964,6 @@
 # ********************
 
 PYGTK_REQUIRED=2.10.0
-GNOME_PYTHON_DESKTOP_REQUIRED=2.17.3
 
 if test "$with_platform" = "hildon" -a "$with_platform_variant" = "maemo3"; then
 GTK_REQUIRED=2.6.0
@@ -21431,12 +21430,15 @@
     pkg_cv_GNOME_CFLAGS="$GNOME_CFLAGS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gconf-2.0 >= \$GCONF_REQUIRED\"") >&5
-  ($PKG_CONFIG --exists --print-errors "gconf-2.0 >= $GCONF_REQUIRED") 2>&5
+    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gconf-2.0 >= \$GCONF_REQUIRED
+                             launchpad-integration\"") >&5
+  ($PKG_CONFIG --exists --print-errors "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration") 2>&5
   ac_status=$?
   $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_GNOME_CFLAGS=`$PKG_CONFIG --cflags "gconf-2.0 >= $GCONF_REQUIRED" 2>/dev/null`
+  pkg_cv_GNOME_CFLAGS=`$PKG_CONFIG --cflags "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -21447,12 +21449,15 @@
     pkg_cv_GNOME_LIBS="$GNOME_LIBS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gconf-2.0 >= \$GCONF_REQUIRED\"") >&5
-  ($PKG_CONFIG --exists --print-errors "gconf-2.0 >= $GCONF_REQUIRED") 2>&5
+    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gconf-2.0 >= \$GCONF_REQUIRED
+                             launchpad-integration\"") >&5
+  ($PKG_CONFIG --exists --print-errors "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration") 2>&5
   ac_status=$?
   $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_GNOME_LIBS=`$PKG_CONFIG --libs "gconf-2.0 >= $GCONF_REQUIRED" 2>/dev/null`
+  pkg_cv_GNOME_LIBS=`$PKG_CONFIG --libs "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -21470,14 +21475,17 @@
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        GNOME_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "gconf-2.0 >= $GCONF_REQUIRED" 2>&1`
+	        GNOME_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration" 2>&1`
         else
-	        GNOME_PKG_ERRORS=`$PKG_CONFIG --print-errors "gconf-2.0 >= $GCONF_REQUIRED" 2>&1`
+	        GNOME_PKG_ERRORS=`$PKG_CONFIG --print-errors "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$GNOME_PKG_ERRORS" >&5
 
-	{ { $as_echo "$as_me:$LINENO: error: Package requirements (gconf-2.0 >= $GCONF_REQUIRED) were not met:
+	{ { $as_echo "$as_me:$LINENO: error: Package requirements (gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration) were not met:
 
 $GNOME_PKG_ERRORS
 
@@ -21488,7 +21496,8 @@
 and GNOME_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details.
 " >&5
-$as_echo "$as_me: error: Package requirements (gconf-2.0 >= $GCONF_REQUIRED) were not met:
+$as_echo "$as_me: error: Package requirements (gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration) were not met:
 
 $GNOME_PKG_ERRORS
 
@@ -22411,17 +22420,14 @@
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
     { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"
-    pygtk-2.0 >= \$PYGTK_REQUIRED
-    gnome-python-desktop-2.0 >= \$GNOME_PYTHON_DESKTOP_REQUIRED\"") >&5
+    pygtk-2.0 >= \$PYGTK_REQUIRED\"") >&5
   ($PKG_CONFIG --exists --print-errors "
-    pygtk-2.0 >= $PYGTK_REQUIRED
-    gnome-python-desktop-2.0 >= $GNOME_PYTHON_DESKTOP_REQUIRED") 2>&5
+    pygtk-2.0 >= $PYGTK_REQUIRED") 2>&5
   ac_status=$?
   $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
   pkg_cv_PYGTK_CFLAGS=`$PKG_CONFIG --cflags "
-    pygtk-2.0 >= $PYGTK_REQUIRED
-    gnome-python-desktop-2.0 >= $GNOME_PYTHON_DESKTOP_REQUIRED" 2>/dev/null`
+    pygtk-2.0 >= $PYGTK_REQUIRED" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -22433,17 +22439,14 @@
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
     { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"
-    pygtk-2.0 >= \$PYGTK_REQUIRED
-    gnome-python-desktop-2.0 >= \$GNOME_PYTHON_DESKTOP_REQUIRED\"") >&5
+    pygtk-2.0 >= \$PYGTK_REQUIRED\"") >&5
   ($PKG_CONFIG --exists --print-errors "
-    pygtk-2.0 >= $PYGTK_REQUIRED
-    gnome-python-desktop-2.0 >= $GNOME_PYTHON_DESKTOP_REQUIRED") 2>&5
+    pygtk-2.0 >= $PYGTK_REQUIRED") 2>&5
   ac_status=$?
   $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
   pkg_cv_PYGTK_LIBS=`$PKG_CONFIG --libs "
-    pygtk-2.0 >= $PYGTK_REQUIRED
-    gnome-python-desktop-2.0 >= $GNOME_PYTHON_DESKTOP_REQUIRED" 2>/dev/null`
+    pygtk-2.0 >= $PYGTK_REQUIRED" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -22462,12 +22465,10 @@
 fi
         if test $_pkg_short_errors_supported = yes; then
 	        PYGTK_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "
-    pygtk-2.0 >= $PYGTK_REQUIRED
-    gnome-python-desktop-2.0 >= $GNOME_PYTHON_DESKTOP_REQUIRED" 2>&1`
+    pygtk-2.0 >= $PYGTK_REQUIRED" 2>&1`
         else
 	        PYGTK_PKG_ERRORS=`$PKG_CONFIG --print-errors "
-    pygtk-2.0 >= $PYGTK_REQUIRED
-    gnome-python-desktop-2.0 >= $GNOME_PYTHON_DESKTOP_REQUIRED" 2>&1`
+    pygtk-2.0 >= $PYGTK_REQUIRED" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$PYGTK_PKG_ERRORS" >&5
@@ -22495,8 +22496,8 @@
 fi
 
 if test "$enable_python" = "no" -a "$need_python" = "yes"; then
-  { { $as_echo "$as_me:$LINENO: error: Some games need python, but python, pygtk or gnome-python-desktop packages were not found." >&5
-$as_echo "$as_me: error: Some games need python, but python, pygtk or gnome-python-desktop packages were not found." >&2;}
+  { { $as_echo "$as_me:$LINENO: error: Some games need python, but python or pygtk packages were not found." >&5
+$as_echo "$as_me: error: Some games need python, but python or pygtk packages were not found." >&2;}
    { (exit 1); exit 1; }; }
 fi
 
@@ -23685,7 +23686,7 @@
 do
   for j in $libggz_header;
   do
-    echo "configure: 23688: $i/$j" >&5
+    echo "configure: 23689: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       libggz_incdir=$i
@@ -23959,7 +23960,7 @@
 do
   for j in $ggzmod_header;
   do
-    echo "configure: 23962: $i/$j" >&5
+    echo "configure: 23963: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       ggzmod_incdir=$i
@@ -24099,7 +24100,7 @@
 do
   for j in ggz-config;
   do
-    echo "configure: 24102: $i/$j" >&5
+    echo "configure: 24103: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       ggz_config_dir=$i
@@ -24317,7 +24318,7 @@
 do
   for j in $libggz_header;
   do
-    echo "configure: 24320: $i/$j" >&5
+    echo "configure: 24321: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       libggz_incdir=$i
@@ -24607,7 +24608,7 @@
 do
   for j in $ggzdmod_header;
   do
-    echo "configure: 24610: $i/$j" >&5
+    echo "configure: 24611: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       ggzdmod_incdir=$i
@@ -25722,7 +25723,7 @@
 do
   for j in $gnutls_header;
   do
-    echo "configure: 25725: $i/$j" >&5
+    echo "configure: 25726: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       gnutls_incdir=$i
@@ -25813,7 +25814,7 @@
 do
   for j in openssl/ssl.h;
   do
-    echo "configure: 25816: $i/$j" >&5
+    echo "configure: 25817: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       ssl_incdir=$i
