Index: gnome-games-2.27.5/configure
===================================================================
--- gnome-games-2.27.5.orig/configure	2009-07-28 10:18:09.000000000 +1000
+++ gnome-games-2.27.5/configure	2009-07-28 10:19:22.000000000 +1000
@@ -21810,12 +21810,15 @@
         pkg_cv_GNOME_CFLAGS="$GNOME_CFLAGS"
     else
         if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gconf-2.0 >= \$GCONF_REQUIRED\"") >&5
-  ($PKG_CONFIG --exists --print-errors "gconf-2.0 >= $GCONF_REQUIRED") 2>&5
+    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gconf-2.0 >= \$GCONF_REQUIRED
+                             launchpad-integration\"") >&5
+  ($PKG_CONFIG --exists --print-errors "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration") 2>&5
   ac_status=$?
   $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_GNOME_CFLAGS=`$PKG_CONFIG --cflags "gconf-2.0 >= $GCONF_REQUIRED" 2>/dev/null`
+  pkg_cv_GNOME_CFLAGS=`$PKG_CONFIG --cflags "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -21828,12 +21831,15 @@
         pkg_cv_GNOME_LIBS="$GNOME_LIBS"
     else
         if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gconf-2.0 >= \$GCONF_REQUIRED\"") >&5
-  ($PKG_CONFIG --exists --print-errors "gconf-2.0 >= $GCONF_REQUIRED") 2>&5
+    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"gconf-2.0 >= \$GCONF_REQUIRED
+                             launchpad-integration\"") >&5
+  ($PKG_CONFIG --exists --print-errors "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration") 2>&5
   ac_status=$?
   $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_GNOME_LIBS=`$PKG_CONFIG --libs "gconf-2.0 >= $GCONF_REQUIRED" 2>/dev/null`
+  pkg_cv_GNOME_LIBS=`$PKG_CONFIG --libs "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -21852,14 +21858,17 @@
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        GNOME_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "gconf-2.0 >= $GCONF_REQUIRED"`
+	        GNOME_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration"`
         else
-	        GNOME_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "gconf-2.0 >= $GCONF_REQUIRED"`
+	        GNOME_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration"`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$GNOME_PKG_ERRORS" >&5
 
-	{ { $as_echo "$as_me:$LINENO: error: Package requirements (gconf-2.0 >= $GCONF_REQUIRED) were not met:
+	{ { $as_echo "$as_me:$LINENO: error: Package requirements (gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration) were not met:
 
 $GNOME_PKG_ERRORS
 
@@ -21870,7 +21879,8 @@
 and GNOME_LIBS to avoid the need to call pkg-config.
 See the pkg-config man page for more details.
 " >&5
-$as_echo "$as_me: error: Package requirements (gconf-2.0 >= $GCONF_REQUIRED) were not met:
+$as_echo "$as_me: error: Package requirements (gconf-2.0 >= $GCONF_REQUIRED
+                             launchpad-integration) were not met:
 
 $GNOME_PKG_ERRORS
 
@@ -24227,7 +24237,7 @@
 do
   for j in $libggz_header;
   do
-    echo "configure: 24230: $i/$j" >&5
+    echo "configure: 24240: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       libggz_incdir=$i
@@ -24501,7 +24511,7 @@
 do
   for j in $ggzcore_header;
   do
-    echo "configure: 24504: $i/$j" >&5
+    echo "configure: 24514: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       ggzcore_incdir=$i
@@ -24674,7 +24684,7 @@
 do
   for j in $ggzmod_header;
   do
-    echo "configure: 24677: $i/$j" >&5
+    echo "configure: 24687: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       ggzmod_incdir=$i
@@ -24814,7 +24824,7 @@
 do
   for j in ggz-config;
   do
-    echo "configure: 24817: $i/$j" >&5
+    echo "configure: 24827: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       ggz_config_dir=$i
@@ -25032,7 +25042,7 @@
 do
   for j in $libggz_header;
   do
-    echo "configure: 25035: $i/$j" >&5
+    echo "configure: 25045: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       libggz_incdir=$i
@@ -25322,7 +25332,7 @@
 do
   for j in $ggzdmod_header;
   do
-    echo "configure: 25325: $i/$j" >&5
+    echo "configure: 25335: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       ggzdmod_incdir=$i
@@ -26437,7 +26447,7 @@
 do
   for j in $gnutls_header;
   do
-    echo "configure: 26440: $i/$j" >&5
+    echo "configure: 26450: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       gnutls_incdir=$i
@@ -26528,7 +26538,7 @@
 do
   for j in openssl/ssl.h;
   do
-    echo "configure: 26531: $i/$j" >&5
+    echo "configure: 26541: $i/$j" >&5
     if test -r "$i/$j"; then
       echo "taking that" >&5
       ssl_incdir=$i
